src/app/components-new/authentication/MnemonicInput.vue

```diff
@@ -61,6 +61,7 @@ div
                 const value = this.mnemonicFields
                     .map(field => field.value.trim().toLocaleLowerCase())
                     .join(' ')
+                    .trim()
 
                 // Allows use of v-model
                 this.$emit('input', value)
```

Comment:
Here and in general, I would avoid two-way bindings and update local variables only when necessary (e.g. at the point of validation).

Reply:
Why though? 2 way binding is the vue way of doing things, this allows you to use the component with v-model the same way you would use any input field.

It doesn't implement a full 2 way binding currently though, it will only take the value from the outside when the component is created.

Reply:
It will also update it on any and every input change, which isn't necessary (unless you want to perform real-time validation or some other action, which you're not doing here). If there's no reason to do it, why do it?

Reply:
Using references is considered bad practice in Vue, because it makes the parent component tightly coupled with the child component, and should only be done when the standard ways of binding data fail. This is also true for native HTML elements - consider what happens when you need to change an input field to some custom component in the future. With a v-model binding, this change is seamless.

See https://vuejs.org/v2/guide/components-edge-cases.html#Accessing-Child-Component-Instances-amp-Child-Elements 

> $refs are only populated after the component has been rendered, and they are not reactive. It is only meant as an escape hatch for direct child manipulation

and https://vuejs.org/v2/style-guide/#Implicit-parent-child-communication-use-with-caution

> An ideal Vue application is props down, events up